TopMovies/Core/Action/MoviesAction.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  MoviesAction.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 21.02.2021.
+//
+
+import ReSwift
+
+struct MoviesRequestAction: Action {
+  let categoryId: MovieCategory.ID
+  let requestType: RequestType
+  
+  enum RequestType { case reload, loadMore }
+}
+
+struct MoviesResponseAction: Action {
+  let categoryId: MovieCategory.ID
+  let list: [Movie]
+  let requestedType: MoviesRequestAction.RequestType
+  let responseType: ResponseType
+  let nextPage: Int?
+  
+  enum ResponseType {
+    case downloading, completed
+    case failed(error: Error)
+  }
```

Comment:
You will have to pass some default values or invalid ones if you will try to instantiate that action, for example for downloading case.

So you should have 4 different actions:

```swift
struct RequestedMoviesListAction: Action { /* id, type */ }
struct DownloadingMoviesListAction: Action { /* id, type */ }
struct CompletedMoviesListAction: Action { /* id, type, list, nextPage */ }
struct FailedMoviesListAction: Action { /* id, type, error */ }
```